dali/test/python/test_dali_tf_dataset_eager.py

```diff
@@ -60,12 +131,33 @@ def test_tf_dataset_wrong_placement_gpu():
     for sample in dataset:
         pass
 
+@raises(Exception)
+def check_tf_dataset_mismatched_input_type(wrong_input_dataset, wrong_input_name, wrong_input_layout=None):
+    pipe = many_input_pipeline(True, "cpu", None, ["a", "b"], batch_size=8, num_threads=4, device_id=0)
+
+    with tf.device('/cpu:0'):
+        input_dataset = tf.data.Dataset.from_tensors(np.full((2, 2), 42)).repeat()
+        dali_dataset = dali_tf.experimental.DALIDatasetWithInputs(
+                input_datasets=wrong_input_dataset,
+                input_names=wrong_input_name,
+                input_layouts=wrong_input_layout,
+                pipeline=pipe,
+                batch_size=pipe.batch_size,
+                output_shapes=(None, None),
+                output_dtypes=(tf.int32, tf.int32),
+                num_threads=pipe.num_threads,
+                device_id=pipe.device_id)
+        return dali_dataset
+
+def test_tf_dataset_mismatched_input_type():
+    input_dataset = tf.data.Dataset.from_tensors(np.full((2, 2), 42)).repeat()
+    for wrong_input_dataset in ["str", [input_dataset], input_dataset]:
+        for wrong_input_name in [42, ["a"]]:
+            yield check_tf_dataset_mismatched_input_type, wrong_input_dataset, wrong_input_name
```

Comment:
Here you are checking that there's an error when you provide two wrong arguments at the same time. I think we should check one at a time.

Reply:
Done